### PR TITLE
Add tools/triage.sh

### LINKED ISSUE #1237: Create tool to execute the triaging workflow
## Summary

The Editor has a more-or-less formalised triaging workflow described in [docs/TRIAGING.md](https://github.com/xsf/xeps/blob/master/docs/TRIAGING.md#flowchart). Doing these steps manually is tedious and error-prone.

## Work proposal

A tool should be created which, given two git commits (the PR head and the merge base between the PR head and the current main branch), triages the changes contained in that range and provides a machine-readable judgement and human-readable action items.

## Requirements

- MUST be accurate: In case of doubt, the tool should delegate to a human (by providing next steps on the output) instead of guessing.

- MUST provide machine-readable output: This is for future CI integration which may eventually post notices to GitHub or assign labels.

### PR DESCRIPTION
**Pull request by moparisthebest**
Fixes #1237

So for a first go this can be ran manually, or ran by a github action and dumped into a comment, or grep'd for "^tag" and used to automatically apply tags by a github action etc

### CODE REVIEW

File: tools/triage.sh [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        echo "error: multiple protoxeps cannot be created in 1 PR"
+        exit 1
+    fi
+    expected_title="ProtoXEP: $(xpath "$protoxep" '/xep/header/title/text()')"
+    if [ "$pr_title" != "$expected_title" ]
+    then
+        echo "tag: bad-title"
+        echo "error: expected title '$expected_title' got title '$pr_title'"
+        exit 1
+    fi
```
Comment (horazont) [inline tools/triage.sh:69]: I wouldn't want to make this a hard error--that seems just superfluous bureaucracy to me. 

Enforcing the ProtoXEP prefix is useful (though labelling could replace that eventually), but let's not get too nitpicky about the overall title.

File: tools/triage.sh [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    echo "tag: ready-to-merge"
+    echo "manual: be sure type '$(xpath "$protoxep" '/xep/header/type/text()')' is appropriate for XEP content"
+    exit 0
+fi
```
Comment (horazont) [inline tools/triage.sh:73]: Is this handling the case where someone mixes a protoxep submission with other XEP changes?

File: tools/triage.sh [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    
+    status="$(xpath "$xep" '/xep/header/status/text()')"    
+    approvers=""
+    if echo "$status" | grep -E '^(Experimental|Deferred)$' &>/dev/null
+    then
+        # Experimental or Deferred
+        if [ "$status" == 'Deferred' ]
+        then
+            echo "tag: needs-editor-action"
+            echo "manual: move $xep_num status to Experimental when merged"
```
Comment (horazont) [inline tools/triage.sh:120]: The status change should happen in the same commit which writes the revision block.

File: tools/triage.sh [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+if echo "$all_approvers" | grep '^Board$' &>/dev/null
+then
+    echo "tag: needs-board"
+fi
+
+if echo "$all_approvers" | grep '^Unknown$' &>/dev/null
+then
+    echo "tag: needs-unknown-approver"
+    echo "warning: approver is unknown, perhaps old XEP that needs fixed?"
```
Comment (horazont) [inline tools/triage.sh:197]: This should be error/manual, shouldn't it?

**Review by horazont — COMMENTED** [review]
Not fully awake anymore. Mostly looking good I think, some thoughts I left.

May need another round of more careful looking later, but maybe you already have thoughts on those thoughts.

File: tools/triage.sh (hunk 4 above)
Comment (moparisthebest) [reply tools/triage.sh:197]: I'd prefer to just put an `<approver/>` in all old XEPs and remove this

File: tools/triage.sh (hunk 3 above)
Comment (moparisthebest) [reply tools/triage.sh:120]: Currently this doesn't go by commit, it looks at the entire PR as 1 change :/

File: tools/triage.sh (hunk 1 above)
Comment (moparisthebest) [reply tools/triage.sh]: This is just a warning now

File: tools/triage.sh (hunk 2 above)
Comment (moparisthebest) [reply tools/triage.sh:81]: Oops good catch, now mixing these is a hard error

File: tools/triage.sh (hunk 3 above)
Comment (horazont) [reply tools/triage.sh:122]: Right, I meant that this should be part of the note for the editor. "when merged" can easily be read as "change status after merging" which would not be the correct action.

Comment (moparisthebest) [reply tools/triage.sh]: I changed this language, let me know

File: tools/triage.sh (hunk 4 above)
Comment (moparisthebest) [reply tools/triage.sh:199]: 205 XEPs are missing approver :skull:

Comment (horazont) [reply tools/triage.sh:199]: In favour of "error" then.

**Comment by tmolitor-stud-tu** [comment]
What is missing/needed to move this one forward?

**Pull request merged**